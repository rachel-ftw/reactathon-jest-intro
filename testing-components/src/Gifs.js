// require('dotenv').config()
import React, { Component } from 'react'

const giphyAPIKey = process.env.REACT_APP_GIPHY_API_KEY

class Gifs extends Component {
  state = {
    searchTerm: '',
    imageUrl: '',
  }

  handleChange = e => this.setState({ searchTerm: e.target.value })

  handleSubmit = e => {
    e.preventDefault()
    const url = `http://api.giphy.com/v1/gifs/translate?api_key=${giphyAPIKey}&s=${this.state.searchTerm}`

    fetch(url)
      .then(response => response.json())
      .then(payload => {
        let imageUrl = payload.data.images.original.url
        this.setState({ imageUrl })
      })
      .catch(e => console.error(e))
  }

  render() {
    const { imageUrl, searchTerm } = this.state
    return (
      <div>
        <form onSubmit={e => this.handleSubmit(e)}>
          <input
            type="text"
            value={searchTerm}
            onChange={e => this.handleChange(e)}
            placeholder="Search for a Gif" />
          <button type="submit">search</button>
        </form>
        {
          imageUrl.length ?
            <img src={imageUrl} alt="Gif"></img> :
            <div></div>
        }
      </div>
    )
  }
}

export default Gifs
