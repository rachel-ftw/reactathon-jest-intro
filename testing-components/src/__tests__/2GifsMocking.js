import React from 'react'

import Gifs from '../Gifs'

describe('<Gifs /> Mocking', () => {
  it('with snapshot & renderer', () => {

  });
})
