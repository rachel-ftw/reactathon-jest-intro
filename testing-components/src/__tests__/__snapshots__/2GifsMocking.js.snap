// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gifs /> Mocking with snapshot & renderer 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search for a Gif"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      search
    </button>
  </form>
  <div />
</div>
`;

exports[`<Gifs /> with snapshot & renderer 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search for a Gif"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      search
    </button>
  </form>
  <div />
</div>
`;
